core/src/avm1/activation.rs

```diff
@@ -1190,8 +1190,13 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
     }
 
     fn action_get_time(&mut self) -> Result<FrameControl<'gc>, Error<'gc>> {
+        self.context.times_get_time_called += 1;
+        if self.context.times_get_time_called >= 20 && self.context.times_get_time_called % 5 == 0 {
+            *self.context.time_offset += 1;
+        }
+
         let time = self.context.navigator.time_since_launch().as_millis() as u32;
-        self.context.avm1.push(time);
+        self.context.avm1.push(time + *self.context.time_offset);
```

Comment:
Maybe `wrapping_add` here in the admittedly rare case that someone leaves their game running for a month.

Reply:
Oh, it about checks out that `u32` can store around a month worth of millis. But then more things will fail than just `time_offset`.
Rather, this
```
let time = self.context.navigator.time_since_launch().as_millis() as u32;
```
should directly convert to target f64, right?

Reply:
`as_millis()` returns a `u128` which will exhibit wrapping behavior when truncated to `u32`, so I think that's fine in this case. You're right in that there are probably other cases where this blows up.